src/main/kotlin/solve/rendering/engine/scene/Scene.kt

```diff
@@ -1,39 +1,58 @@
 package solve.rendering.engine.scene
 
-import solve.rendering.engine.core.renderers.Renderer
-
-class Scene(renderers: List<Renderer> = emptyList()) {
-    private val _renderObjects = mutableListOf<RenderObject>()
-    val renderObjects: List<RenderObject>
-        get() = _renderObjects
-
-    private val _renderers = renderers
-    val renderers: List<Renderer>
-        get() = _renderers
-
-    fun update(deltaTime: Float) {
-        _renderObjects.forEach { gameObject ->
-            if (gameObject.isDestroyed) {
-                removeRenderObject(gameObject)
-                return@forEach
-            }
-
-            gameObject.update(deltaTime)
-            _renderers.forEach { it.render() }
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import solve.rendering.engine.core.renderers.FramesRenderer
+import solve.rendering.engine.core.renderers.LandmarkLayerRenderer
+import solve.scene.model.Layer
+import java.util.concurrent.CopyOnWriteArrayList
+
+class Scene(val framesRenderer: FramesRenderer) {
+    private val _landmarkRenderers = CopyOnWriteArrayList<LandmarkLayerRenderer>()
+    private val _landmarkRenderersLayers = mutableMapOf<LandmarkLayerRenderer, Layer>()
```

Comment:
can we use LandmarkLayerRenderer as key without declaring equals or hashcode?

Reply:
No :)

Reply:
But in this case we have only a few map elements, so it's not critical. We can't override these functions on `LandmarkLayerRenderer` because all properties of that class are mutable.